### PR TITLE
Implemented types for this project

### PR DESCRIPTION
**Pull request by lucasvictor3**
Implemented types that was already commented on code and migrated all .js files to .ts format.

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/** @flow */
+
+"use strict";
+
+/*::
+type Handler = (event?: any) => void;
```
Comment (lucasvictor3) [inline index.ts]: Faltou eu remover aqui =x

File: index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+type HandlerMap = { [name: string]: Set<Handler>, __proto__: null };
+*/
+
+/* prevent bug that warns about event not beeing used. */
+// eslint-disable-next-line
+type Handler = (event: any) => void;
```
Comment (lucasvictor3) [inline index.ts]: tinha uma issue sobre isso no repo typescript-eslint. Mas alguem fechou porque nao conseguiu reproduzir na master.

**Review by duailibe — CHANGES_REQUESTED** [review]
You don't need to rename the file to .js

Just add a `index.d.ts` next to `index.js` and TS will pick that up

**Review by duailibe — COMMENTED** [review]
Much better =)

Now you have to add `index.d.ts` here so it will be included in the package:

https://github.com/duailibe/unemit/blob/50a0694786c5d18afd3686520e8d4338af417e15/package.json#L14-L16

**Review by duailibe — CHANGES_REQUESTED** [review]
What I said above

**Review by duailibe — APPROVED** [review]

**Pull request merged**